### ISSUE TITLE
Fix interface between members_to_percentiles and ECC

### ISSUE THREAD

**Issue by gavinevans** [metoppv/improver#232]
As an IMPROVER developer, I want the `members_to_percentiles` and ECC plugins to function together so that the input and output cubes have the correct metadata.

Related issues: #148 

The `members_to_percentiles` plugin can output a cube with both a `realization` dimension coordinate and a `realization` scalar coordinate. This causes a problem when rebadging the `percentiles` coordinate to be a `realization` coordinate.

Acceptance criteria:
 * Ensure `members_to_percentiles` outputs a cube with sensible metadata.
 * Ensure that ECC is robust if a cube with two `realization` coordinates is passed in.
 * Unit tests

**Comment by gavinevans** [comment]
I don't think that it should be possible to have a `percentiles_over_realization` and a `realization` coordinate on the same cube, although it would be possible to have other forms of a percentile coordinate e.g. `percentiles_over_time` and a `realization` coordinate on the same cube.

**Comment by arh89** [comment]
Note: Following ECC, the `time` scalar coordinate is promoted to a dimension coordinate (with `forecast_period` and `forecast_reference_time` also 'promoted' from scalar to auxiliary coordinates).

Example:
level_1 output:
`
wind_speed / (m s-1)                (realization: 12; projection_y_coordinate: 970; projection_x_coordinate: 1042)
     Dimension coordinates:
          realization                           x                            -                             -
          projection_y_coordinate               -                            x                             -
          projection_x_coordinate               -                            -                             x
     Scalar coordinates:
          forecast_period: 0 seconds
          forecast_reference_time: 2017-02-01 03:00:00
          height: 10.0 m
          time: 2017-02-01 03:00:00
     Attributes:
          Conventions: CF-1.5
          grid_id: enukx_standard_v1
          institution: Met Office
          source: Met Office Unified Model
          title: Operational MOGREPS-UK Model Forecast
          um_version: 10.4
`

Following members_to_percentiles and removal of the scalar `realization` coordinate (mentioned in the issue description):
`
wind_speed / (m s-1)                (percentile_over_realization: 15; projection_y_coordinate: 970; projection_x_coordinate: 1042)
     Dimension coordinates:
          percentile_over_realization                           x                            -                             -
          projection_y_coordinate                               -                            x                             -
          projection_x_coordinate                               -                            -                             x
     Scalar coordinates:
          forecast_period: 0 seconds
          forecast_reference_time: 2017-02-01 03:00:00
          height: 10.0 m
          time: 2017-02-01 03:00:00
     Attributes:
          Conventions: CF-1.5
          grid_id: enukx_standard_v1
          institution: Met Office
          source: Met Office Unified Model
          title: Operational MOGREPS-UK Model Forecast
          um_version: 10.
`

And finally, after ECC using rebadging:
`
wind_speed / (m s-1)                (realization: 12; time: 1; projection_y_coordinate: 970; projection_x_coordinate: 1042)
     Dimension coordinates:
          realization                           x         -                           -                             -
          time                                  -         x                           -                             -
          projection_y_coordinate               -         -                           x                             -
          projection_x_coordinate               -         -                           -                             x
     Auxiliary coordinates:
          forecast_period                       -         x                           -                             -
          forecast_reference_time               -         x                           -                             -
     Scalar coordinates:
          height: 10.0 m
     Attributes:
          Conventions: CF-1.5
          grid_id: enukx_standard_v1
          institution: Met Office
          source: Met Office Unified Model
          title: Operational MOGREPS-UK Model Forecast
          um_version: 10.4
`

...I think this treatment of the `time` coordinate may also need checking for correctness.
